test/src/unit-capi-object_mgmt.cc

```diff
@@ -527,115 +379,35 @@ TEST_CASE_METHOD(
   std::string walk_str, ls_str;
   int rc;
 
-  if (supports_s3_) {
-    // S3
-    remove_temp_dir(S3_TEMP_DIR);
-    create_hierarchy(S3_TEMP_DIR);
-    golden_walk = get_golden_walk(S3_TEMP_DIR);
-    golden_ls = get_golden_ls(S3_TEMP_DIR);
-    walk_str.clear();
-    ls_str.clear();
-    rc = tiledb_object_walk(
-        ctx_, S3_TEMP_DIR.c_str(), TILEDB_PREORDER, write_path, &walk_str);
-    CHECK(rc == TILEDB_OK);
-    rc = tiledb_object_walk(
-        ctx_, S3_TEMP_DIR.c_str(), TILEDB_POSTORDER, write_path, &walk_str);
-    CHECK(rc == TILEDB_OK);
-    CHECK_THAT(golden_walk, Catch::Equals(walk_str));
-    rc = tiledb_object_ls(ctx_, S3_TEMP_DIR.c_str(), write_path, &ls_str);
-    CHECK(rc == TILEDB_OK);
-    CHECK_THAT(golden_ls, Catch::Equals(ls_str));
-    remove_temp_dir(S3_TEMP_DIR);
-  } else if (supports_azure_) {
-    // AZURE
-    remove_temp_dir(AZURE_TEMP_DIR);
-    create_hierarchy(AZURE_TEMP_DIR);
-    golden_walk = get_golden_walk(AZURE_TEMP_DIR);
-    golden_ls = get_golden_ls(AZURE_TEMP_DIR);
-    walk_str.clear();
-    ls_str.clear();
-    rc = tiledb_object_walk(
-        ctx_, AZURE_TEMP_DIR.c_str(), TILEDB_PREORDER, write_path, &walk_str);
-    CHECK(rc == TILEDB_OK);
-    rc = tiledb_object_walk(
-        ctx_, AZURE_TEMP_DIR.c_str(), TILEDB_POSTORDER, write_path, &walk_str);
-    CHECK(rc == TILEDB_OK);
-    CHECK_THAT(golden_walk, Catch::Equals(walk_str));
-    rc = tiledb_object_ls(ctx_, AZURE_TEMP_DIR.c_str(), write_path, &ls_str);
-    CHECK(rc == TILEDB_OK);
-    CHECK_THAT(golden_ls, Catch::Equals(ls_str));
-    remove_temp_dir(AZURE_TEMP_DIR);
-  } else if (supports_hdfs_) {
-    // HDFS
-    remove_temp_dir(HDFS_TEMP_DIR);
-    create_hierarchy(HDFS_TEMP_DIR);
-    golden_walk = get_golden_walk(HDFS_FULL_TEMP_DIR);
-    golden_ls = get_golden_ls(HDFS_FULL_TEMP_DIR);
-    walk_str.clear();
-    ls_str.clear();
-    rc = tiledb_object_walk(
-        ctx_, HDFS_TEMP_DIR.c_str(), TILEDB_PREORDER, write_path, &walk_str);
-    CHECK(rc == TILEDB_OK);
-    rc = tiledb_object_walk(
-        ctx_, HDFS_TEMP_DIR.c_str(), TILEDB_POSTORDER, write_path, &walk_str);
-    CHECK(rc == TILEDB_OK);
-    CHECK_THAT(golden_walk, Catch::Equals(walk_str));
-    rc = tiledb_object_ls(ctx_, HDFS_TEMP_DIR.c_str(), write_path, &ls_str);
-    CHECK(rc == TILEDB_OK);
-    CHECK_THAT(golden_ls, Catch::Equals(ls_str));
-    remove_temp_dir(HDFS_TEMP_DIR);
-  } else {
-    // File
-    remove_temp_dir(FILE_FULL_TEMP_DIR);
-    create_hierarchy(FILE_FULL_TEMP_DIR);
-#ifdef _WIN32
-    // `VFS::ls(...)` returns `file:///` URIs instead of Windows paths.
-    golden_walk =
-        get_golden_walk(tiledb::sm::Win::uri_from_path(FILE_FULL_TEMP_DIR));
-    golden_ls =
-        get_golden_ls(tiledb::sm::Win::uri_from_path(FILE_FULL_TEMP_DIR));
-#else
-    golden_walk = get_golden_walk(FILE_FULL_TEMP_DIR);
-    golden_ls = get_golden_ls(FILE_FULL_TEMP_DIR);
-#endif
```

Comment:
@bekadavis9 I believe this is the source of the unit test failure. I'm not familiar with this test, but it looks like we have custom Windows logic here. This isn't your fault, this is an existing hack. I think it would be appropriate to use this same sort of `#ifdef _WIN32` logic in your refactor.